pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2AuthnResponseValidator.java

```diff
@@ -621,7 +617,23 @@ protected void validateAuthenticationStatements(final List<AuthnStatement> authn
                     throw new SAMLAuthnSessionCriteriaException("Authentication session between IDP and subject has ended");
                 }
             }
-            // TODO implement authnContext validation
+            authnClassRefs.add(statement.getAuthnContext().getAuthnContextClassRef().getURI());
+        }
+
+        validateAuthnContextClassRefs(authnClassRefs);
+    }
+
+    protected void validateAuthnContextClassRefs(final List<String> providedAuthnContextClassRefs) {
+        if (!saml2Configuration.getAuthnContextClassRefs().isEmpty()) {
+            logger.debug("Required authentication context class refs are {}", saml2Configuration.getAuthnContextClassRefs());
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Class names should start with an upper case letter (NM_CLASS_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-should-start-with-an-upper-case-letter-nm-class-naming-convention).
Class names should be nouns, in mixed case with the first letter of each internal word capitalized. Try to keep your class names simple and descriptive. Use whole words-avoid acronyms and abbreviations (unless the abbreviation is much more widely used than the long form, such as URL or HTML).